kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -719,3 +861,55 @@ class SnapshotManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
     assert(!logSegmentOpt.isPresent())
   }
 }
+
+class MockSidecarParquetHandler(sidecars: Seq[FileStatus])
+  extends BaseMockParquetHandler
+    with VectorTestUtils {
+  override def readParquetFiles(
+    fileIter: CloseableIterator[FileStatus],
+    physicalSchema: StructType,
+    predicate: Optional[Predicate]): CloseableIterator[ColumnarBatch] = {
+
+    Utils.singletonCloseableIterator(
+      new ColumnarBatch {
+        override def getSchema: StructType = SidecarFile.READ_SCHEMA
+
+        override def getColumnVector(ordinal: Int): ColumnVector = {
+          ordinal match {
+            case 0 => stringVector(sidecars.map(_.getPath)) // path
+            case 1 => longVector(sidecars.map(_.getSize): _*) // size
+            case 2 =>
+              longVector(sidecars.map(_.getModificationTime): _*); // modification time
+          }
+        }
+
+        override def getSize: Int = sidecars.length
+      })
+  }
+}
+
+class MockSidecarJsonHandler(sidecars: Seq[FileStatus])
+  extends BaseMockJsonHandler
+    with VectorTestUtils {
+  override def readJsonFiles(
+    fileIter: CloseableIterator[FileStatus],
+    physicalSchema: StructType,
+    predicate: Optional[Predicate]): CloseableIterator[ColumnarBatch] = {
+
+    Utils.singletonCloseableIterator(
+      new ColumnarBatch {
+        override def getSchema: StructType = SidecarFile.READ_SCHEMA
+
+        override def getColumnVector(ordinal: Int): ColumnVector = {
+          ordinal match {
+            case 0 => stringVector(sidecars.map(_.getPath)) // path
+            case 1 => longVector(sidecars.map(_.getSize): _*) // size
+            case 2 =>
+              longVector(sidecars.map(_.getModificationTime): _*); // modification time
+          }
+        }
+
+        override def getSize: Int = sidecars.length
+      })
```

Comment:
this is a duplicate of the same code as above in mock ParquetHandler.

Reply:
Factored out.